tests/integration/deploy/test_deploy_command.py

```diff
@@ -1285,3 +1285,56 @@ def test_deploy_logs_warning_with_cdk_project(self, template_file):
         deploy_process_execute = run_command(deploy_command_list)
         self.assertIn(warning_message, deploy_process_execute.stdout)
         self.assertEqual(deploy_process_execute.process.returncode, 0)
+
+    @parameterized.expand(["aws-serverless-function.yaml"])
+    def test_update_stack_correct_stack_outputs(self, template):
+        template_path = self.test_data_path.joinpath(template)
+
+        stack_name = self._method_to_stack_name(self.id())
+        self.stacks.append({"name": stack_name})
+
+        # Deploy template that creates single resource
+        deploy_command_list = self.get_deploy_command_list(
+            template_file=template_path,
+            stack_name=stack_name,
+            capabilities="CAPABILITY_IAM",
+            s3_prefix=self.s3_prefix,
+            s3_bucket=self.s3_bucket.name,
+            image_repository=self.ecr_repo_name,
+            force_upload=True,
+            notification_arns=self.sns_arn,
+            parameter_overrides="Parameter=Clarity",
+            kms_key_id=self.kms_key,
+            no_execute_changeset=False,
+            tags="integ=true clarity=yes foo_bar=baz",
+            confirm_changeset=False,
+        )
+
+        deploy_process_execute = run_command(deploy_command_list)
+        self.assertEqual(deploy_process_execute.process.returncode, 0)
+
+        # Deploy template that modifies existing resource, this should only UPDATE
+        template_path = self.test_data_path.joinpath("aws-serverless-function-cdk.yaml")
```

Comment:
why a cdk template? I'm assuming this could be any different template that's not the same template as before, lets make it so its explicit. something like "aws-serverless-function-changed.yaml"

Otherwise the PR looks good to me.

Reply:
Yup, it just has to be a different template file to trigger an update. The CDK template was used here since it happens to use the same resource name.